activerecord/lib/active_record/reflection.rb

```diff
@@ -623,7 +623,11 @@ def calculate_constructable(macro, options)
         # +nil+.
         def inverse_name
           options.fetch(:inverse_of) do
-            @automatic_inverse_of ||= automatic_inverse_of
+            if @automatic_inverse_of == false
```

Comment:
Note that `automatic_inverse_of` can return `false` as well. I'm assuming `nil` is the right return value here as per the comment above.